src/kakarot/precompiles/modexp.cairo

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+
+%lang starknet
+
+// Starkware dependencies
+from starkware.cairo.common.cairo_builtins import HashBuiltin, BitwiseBuiltin
+from starkware.cairo.common.uint256 import Uint256
+
+// Internal dependencies
+from utils.utils import Helpers
+from utils.modexp.modexp_utils import ModExpHelpers
+
+// @title ModExp Precompile related functions.
+// @notice This file contains the logic required to run the modexp precompile
+// @author @dragan2234
+// @custom:namespace PrecompileModExp
+namespace PrecompileModExp {
+    const PRECOMPILE_ADDRESS = 0x05;
+    const MOD_EXP_BYTES_LEN = 32;
+
+    // @notice Run the precompile.
+    // @param input_len The length of input array.
+    // @param input The input array.
+    // @return The output length, output array, and gas usage of precompile.
+    func run{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(_address: felt, input_len: felt, input: felt*) -> (
+        output_len: felt, output: felt*, gas_used: felt
+    ) {
+        alloc_locals;
+        let b_size: Uint256 = Helpers.bytes32_to_uint256(input);
+        let e_size: Uint256 = Helpers.bytes32_to_uint256(input + MOD_EXP_BYTES_LEN);
+        let m_size: Uint256 = Helpers.bytes32_to_uint256(input + MOD_EXP_BYTES_LEN * 2);
+        let b: Uint256 = Helpers.bytes_i_to_uint256(input + MOD_EXP_BYTES_LEN * 3, b_size.low);
+        let e: Uint256 = Helpers.bytes_i_to_uint256(
+            input + MOD_EXP_BYTES_LEN * 3 + b_size.low, e_size.low
+        );
+        let m: Uint256 = Helpers.bytes_i_to_uint256(
+            input + MOD_EXP_BYTES_LEN * 3 + b_size.low + e_size.low, m_size.low
+        );
```

Comment:
how can you make sure that `.low` is enough? in other words that all numbers `_n_ < 2**128`?

Reply:
@ClementWalter since this implementation only support uint256, byte size of `uint256` is never more than 32